flask_project/routes.py

```diff
@@ -1,6 +1,21 @@
+from flask import render_template, request
 from flask_project import app
-from flask import render_template
+from flask_project.url_scanner import IPQS, get_domain, extract_urls
 
 @app.route('/')
 def home_page():
     return render_template('base.html')
+
+@app.route("/scan-text-urls", methods=["POST"])
+def scan_text_urls():
+    """
+    this endpoint extracts all URLs from the text and performs a malicious URL scan on each URL using IPQS(). 
+    The results of the scan are returned as a list of dictionaries.
+    """
+    text = request.json["text"]
```

Comment:
Please use .get method of dictionary and also please add heandling for the case of "text" key is missed